fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchPerformaceTest.java

```diff
@@ -0,0 +1,488 @@
+/*
+ * (C) Copyright IBM Corp. 2017,2019
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.server.test;
+
+import static com.ibm.fhir.model.type.String.string;
+import static com.ibm.fhir.model.type.Uri.uri;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import static org.testng.Assert.assertTrue;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.core.FHIRMediaType;
+import com.ibm.fhir.model.resource.Bundle;
+import com.ibm.fhir.model.resource.Observation;
+import com.ibm.fhir.model.resource.Patient;
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.model.type.Coding;
+import com.ibm.fhir.model.type.code.AdministrativeGender;
+import com.ibm.fhir.model.util.FHIRUtil;
+
+public class SearchPerformaceTest extends FHIRServerTestBase {
+
+    private static final boolean DEBUG_SEARCH = false;
+    private String patientId;
+    private String observationId;
+    private Boolean compartmentSearchSupported = null;
+    // Controls which tenant to use for the test.
+    private final String tanantName = "perform2";
+    // Controls how many observations and patients to create for the test.
+    // Using invocationCount of testng can cause the testng report grows too big, so
+    // this config is used to make sure all test users are created in one testng step.
+    private final int numOfPatientObservationsToCreate = 100;
+
+    /**
+     * Retrieve the server's conformance statement to determine the status of certain runtime options.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void retrieveConfig() throws Exception {
+        compartmentSearchSupported = isComparmentSearchSupported();
+        System.out.println("Compartment-based search supported?: "
+                + compartmentSearchSupported.toString());
+    }
+
+    @Test(groups = { "server-search-performance" }, invocationCount = 1)
+    public void testCreatePatientAndObservation() throws Exception {
+        WebTarget target = getWebTarget();
+        for (int i = 0; i < numOfPatientObservationsToCreate; i++) {
+            // Build a new Patient and then call the 'create' API.
```

Comment:
Is this for the same patient? or separate patients?

Reply:
this will create different observations and patients with different ids, but actually only the last patient id will be used in some of the search tests which uses patient id.

Reply:
great